### PR TITLE
Adding versionless query api to FeatureRepository

### PR DESCRIPTION
**Pull request by cbridgha**
Adding API to FeatureRepository

1) public String getVersionlessFeatureVersionForPlatform(String versionlessFeatureName, String platformName)

2) public Set<String> getCommonPlatformsForFeatureSet(Set<String> features)

Example usage:
FeatureRepository repo = featureRepository;
Set<String> features = new HashSet<>(Arrays.asList("servlet-6.0", "batch-2.1"));
Set<String> commonPlatforms = repo.getCommonPlatformsForFeatureSet(features);
String featureShortName = repo.getVersionlessFeatureVersionForPlatform("jsp","jakartaee-10");

### CODE REVIEW

**Comment by cbridgha** [comment]
#build

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java [hunk 1]
```diff
@@ -417,6 +418,108 @@ private void readCache(boolean firstInit) {
...
+     * @return ProvisioningFeatureDefinition
+     */
+    private ProvisioningFeatureDefinition getVersionedFeature(String versionlessLinkingFeatureName) {
+        ProvisioningFeatureDefinition result = null;
+        ProvisioningFeatureDefinition feature = getFeature(versionlessLinkingFeatureName);
+        if (feature != null) {
+            //This is the versionless linking feature pointing to a public versioned feature
+            for (FeatureResource versionedFeature : feature.getConstituents(null)) {
+                //There should be only one element - set the result
+                result = getFeature(versionedFeature.getSymbolicName());
```
Comment (rsherget) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java]: There will be two elements for microprofile linking features. They contain both the mpVersion feature and the versioned feature. Adding the below snippet would fix that.
```
ProvisioningFeatureDefinition versionedFeatureDef = getFeature(versionedFeature.getSymbolicName());
if(versionedFeature.getVisibility() != Visibility.PUBLIC){
    continue;
}
result = versionedFeatureDef;
```

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java [hunk 2]
```diff
@@ -417,6 +418,108 @@ private void readCache(boolean firstInit) {
...
 
+    /**
+     * Answer the versioned feature shortName if exists for the passed platform version or null if doesn't exist.
+     *
+     * @return String versionedFeatureName
+     */
+    public String getVersionlessFeatureVersionForPlatform(String versionlessFeatureName, String platformName) {
+        SubsystemFeatureDefinitionImpl versionlessFeature;
+        Map features = getAllFeatures();
+        versionlessFeature = (SubsystemFeatureDefinitionImpl) features.get("io.openliberty.versionless." + versionlessFeatureName);
```
Comment (rsherget) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java]: May be able to just use `getFeature(versionlessFeatureName)` it checks both the symbolic name and the short name if its a public feature.

Comment (cbridgha) [reply dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java]: Thanks! - Yes! - In fact I use that api later on....  haha - change made....

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java (hunk 1 above)
Comment (cbridgha) [reply dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java]: Fixed up...

**Comment by cbridgha** [comment]
#build

**Review by rsherget — APPROVED** [review]
LGTM

**Review by rsherget — APPROVED** [review]
LGTM

**Comment by cbridgha** [comment]
#libby

**Review by hlhoots — APPROVED** [review]
LGTM

**Pull request merged**